DSMR fails to compile on Arduino 2.0.3 + platform 4.4.0: missing hwcrypto/aes.h

### The problem

Using ESPHome with Arduino @ 2.0.3 and platform-espressif32 @ 4.4.0 breaks compilation of the DSMR component. I try to build the project with `esphome run thefile.yaml`, but it fails because it cannot find `hwcrypto/aes.h`.

I installed the development version of ESPHome via `pip install git+https://github.com/esphome/esphome.git@dev` to see if this was recently fixed.

I get the same error with ESPHome 2022.6.1 and any 4.x.x platform. Deleting the `.esphome` and `$HOME/.platformio/.cache` directories doesn't help to fix this.


### Which version of ESPHome has the issue?

2022.7.0-dev

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

n/a

### What platform are you using?

ESP32

### Board

frogboard

### Component causing the issue

DSMR

### Example YAML snippet

```yaml
esp32:
  board: frogboard
  framework:
    type: arduino
    version: 2.0.3
    platform_version: 4.4.0

# ... snip common config, like device name, wifi parameters, etc.

dsmr:
  uart_id: uart_dsmr
  request_interval: 5s

uart:
  id: uart_dsmr
  baud_rate: 115200
  rx_pin:
    number: GPIO21
    inverted: true
  rx_buffer_size: 1700
```


### Anything in the logs that might be useful for us?

```txt
In file included from src/esphome/components/dsmr/dsmr.cpp:6:
.piolibdeps/dsmr/Crypto/AES.h:200:10: fatal error: hwcrypto/aes.h: No such file or directory
 #include "hwcrypto/aes.h"
          ^~~~~~~~~~~~~~~~
compilation terminated.
*** [.pioenvs\dsmr\src\esphome\components\dsmr\dsmr.cpp.o] Error 1
```


### Additional information

The entire build log: [bugreport-build-log.txt](https://github.com/esphome/issues/files/8956063/bugreport-build-log.txt)

--- a/esphome/components/dsmr/__init__.py
+++ b/esphome/components/dsmr/__init__.py
@@ -85,4 +85,4 @@ async def to_code(config):
     cg.add_library("glmnet/Dsmr", "0.5")
 
     # Crypto
-    cg.add_library("rweather/Crypto", "0.2.0")
+    cg.add_library("rweather/Crypto", "0.4.0")

--- a/platformio.ini
+++ b/platformio.ini
@@ -62,7 +62,7 @@ lib_deps =
     mikalhart/TinyGPSPlus@1.0.2                           ; gps
     freekode/TM1651@1.0.1                                 ; tm1651
     glmnet/Dsmr@0.5                                       ; dsmr
-    rweather/Crypto@0.2.0                                 ; dsmr
+    rweather/Crypto@0.4.0                                 ; dsmr
     dudanov/MideaUART@1.1.8                               ; midea
     tonia/HeatpumpIR@1.0.20                               ; heatpumpir
 build_flags =

